lib/services/android-plugin-build-service.ts

```diff
@@ -0,0 +1,365 @@
+import * as path from "path";
+import * as shell from "shelljs";
+
+const xml2js = require("xml2js");
+
+export class AndroidPluginBuildService implements IAndroidPluginBuildService {
+
+	constructor(private $fs: IFileSystem,
+		private $childProcess: IChildProcess,
+		private $hostInfo: IHostInfo,
+		private $androidToolsInfo: IAndroidToolsInfo) { }
+
+	private static ANDROID_MANIFEST_XML = "AndroidManifest.xml";
+	private static INCLUDE_GRADLE = "include.gradle";
+	private static MANIFEST_ROOT = {
+		$: {
+			"xmlns:android": "http://schemas.android.com/apk/res/android"
+		}
+	};
+	private static ANDROID_PLUGIN_GRADLE_TEMPLATE = "gradle-plugin";
+
+	private getAndroidSourceDirectories(source: string): Array<string> {
+		const directories = ["res", "java", "assets", "jniLibs"];
+		const resultArr: Array<string> = [];
+		this.$fs.enumerateFilesInDirectorySync(source, (file, stat) => {
+			if (stat.isDirectory() && _.some(directories, (element) => file.endsWith(element))) {
+				resultArr.push(file);
+				return true;
+			}
+		});
+
+		return resultArr;
+	}
+
+	private getManifest(platformsDir: string) {
```

Comment:
Why we need to traverse the platforms dir to find the `manifest` file. Shoudn't it be in `platforms\android` folder. If this is the case, should'n we just return
```
path.join(platfromsDir, "android", AndroidPluginBuildService.ANDROID_MANIFEST_XML);
```

Reply:
The plugin's platform folder does not necessarily have an `AndroidManifest.xml` file

Reply:
The migration tool started initially with the thought that it could also support flavors of the android libraries, in which cases there could be more than one manifests. Since I later realized that is a somewhat unrealistic expectation (that plugins would have additional flavors), I dropped the idea, but didn't modify the code as it kept working.

Yes, you can indeed use path.join followed by a check if `fs.existsSync()`